Data/Text/Internal/Fusion/Common.hs

```diff
@@ -318,10 +318,13 @@ lengthI (Stream next s0 _len) = loop_length 0 s0
 -- greater than the number or if the stream can't possibly be as long
 -- as the number supplied, and hence be more efficient.
 compareLengthI :: Integral a => Stream Char -> a -> Ordering
-compareLengthI (Stream next s0 len) n =
-    case compareSize len (fromIntegral n) of
-      Just o  -> o
-      Nothing -> loop_cmp 0 s0
+compareLengthI (Stream next s0 len) n
+    -- Note that @len@ tracks code units whereas we want to compare the length
+    -- in code points. Specifically, a stream with hint @len@ may consist of
+    -- anywhere from @len/2@ to @len@ code points.
+  | Just LT <- compareSize (2*len) (fromIntegral n) = LT
```

Comment:
Is this one correct? Since `n` is in code points, and a code point can be 2 code units, won't
`len=6` and `n=1` give `LT` here while it should be `GT`? I would have thought that the original compare just lacked a call to `codePointsSize` over `n`?

Reply:
Oh dear, yes, I believe you are right.